tortoisebeacon/tortoise_beacon.go

```diff
@@ -364,16 +366,22 @@ func (tb *TortoiseBeacon) handleLayer(ctx context.Context, layer types.LayerID)
 
 	tb.seenEpochsMu.Unlock()
 
-	tb.Log.With().Debug("tortoise beacon got tick",
+	tb.Log.With().Debug("tortoise beacon got tick, waiting until other nodes have the same epoch",
 		log.Uint64("layer", uint64(layer)),
 		log.Uint64("epoch_id", uint64(epoch)))
 
-	tb.handleEpoch(ctx, epoch)
+	epochStartTimer := time.NewTimer(tb.proposalDuration + tb.gracePeriodDuration)
+	defer epochStartTimer.Stop()
+
+	select {
+	case <-tb.CloseChannel():
+	case <-epochStartTimer.C:
+		tb.handleEpoch(ctx, epoch)
+	}
```

Comment:
Based on [the docs](https://golang.org/pkg/time/#Timer.Stop), there's no need to stop a timer in case it already fired. Additionally, if you do have to stop it, the channel should be drained.

```suggestion

	select {
	case <-tb.CloseChannel():
		if !epochStartTimer.Stop() {
			<-epochStartTimer.C
		}
	case <-epochStartTimer.C:
		tb.handleEpoch(ctx, epoch)
	}
```